### DISCUSSION TITLE
How do you deal with resource dependencies in combination with custom destructors?

### DISCUSSION THREAD

**Discussion by codecnotsupported** [bevyengine/bevy#4074] [Q&A]
Suppose you have, resource A and B.
B is derived from A. / Created with resources from A. (Not really relevant)
B must be destroyed before A.
B requires a custom destructor to properly de-allocate itself, and requires access to resource A to do so.

The only way I can make this work (that I can think of) by having a reference of resource A inside of resource B.
But the borrow checker makes this a tad bit difficult, because (as far as I understand, the lifetime of a resource is dynamic).

Is the resource drop order the inverse of the initialization order, because, if that is not then there is the issue that A might be dropped before B.

**Comment by james7132** [comment]
The ideal solution is not to have those dependencies at all or keep both in the same resource and properly implement `Drop` on the combined resource. Do these really need to be separate?

**Reply by codecnotsupported** [reply]
Kind of, I was planning on having a few system that query resources like B.
If I were to put resource B in A I might lose out on parallelization, since A might get mutably accessed.
